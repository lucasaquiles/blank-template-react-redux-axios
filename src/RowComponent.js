import React, { Component } from 'react';

export default class RowComponent extends Component {
    
    render(){
        
        {console.log("obj", this)}
        return (
            <tr key={this.props.key}>
                  <td><img src={this.props.info.avatar} /></td>
                  <td>{this.props.info.first_name}</td>
                  <td>{this.props.info.last_name}</td>
            </tr>
            
        )
    }
}
