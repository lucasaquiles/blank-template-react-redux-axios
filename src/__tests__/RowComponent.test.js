import React from 'react';
import { shallow, mount, render } from  'enzyme';
import configureStore from 'redux-mock-store';
import RowComponent from '../RowComponent';
import toJson from 'enzyme-to-json';

const mockStore = configureStore();

const initialState = {

    simpleReducer: {
        result: {
            load: true,
            data:[]
        }
    }
}

const store = mockStore(initialState);

describe('<RowComponent />', () => {

    it('should render row', ()=> {
        
        const info = {"id":1,"first_name":"George","last_name":"Bluth","avatar":"https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"};
        
        const wrapper = mount (<RowComponent info={info} />)
        
        expect(wrapper).toMatchSnapshot();
    });
})