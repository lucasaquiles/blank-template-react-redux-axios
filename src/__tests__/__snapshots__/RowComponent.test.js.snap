// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RowComponent /> should render row 1`] = `
<RowComponent
  info={
    Object {
      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg",
      "first_name": "George",
      "id": 1,
      "last_name": "Bluth",
    }
  }
>
  <tr
    onClick={[Function]}
  >
    <td>
      <img
        src="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
      />
    </td>
    <td>
      George
    </td>
    <td>
      Bluth
    </td>
    <td />
  </tr>
</RowComponent>
`;
